kafka-connect-http/src/main/java/com/github/castorm/kafka/connect/http/HttpSourceTask.java

```diff
@@ -104,11 +117,37 @@ private Offset loadOffset(Map<String, String> initialOffset) {
 
         throttler.throttle(offset.getTimestamp().orElseGet(Instant::now));
 
-        HttpRequest request = requestFactory.createRequest(offset);
+//        HttpRequest request = requestFactory.createRequest(offset);
+
+        List<SourceRecord> records = new ArrayList<>();
+
+        if(handlePagination && !Objects.isNull(modifiedUrl)) {
+            request = HttpRequest.builder()
+                .method(request.getMethod())
+                .url(modifiedUrl)
+                .headers(request.getHeaders())
+                .body(request.getBody())
+                .build();
+        } else {
+            request = requestFactory.createRequest(offset);
+        }
 
         HttpResponse response = execute(request);
 
-        List<SourceRecord> records = responseParser.parse(response);
+        records.addAll(responseParser.parse(response));
+
+        if(handlePagination) {
+            Optional<String> nextPageUrl = responseParser.getNextPageUrl(response);
+            log.info("Next page URL: {}", nextPageUrl.orElse("no value"));
+            if( isNextPageUrlPresent(nextPageUrl) ) {
+                modifiedUrl = appendNextPageUrl
+                    ? baseUrl + nextPageUrl.orElse("")
+                    : nextPageUrl.orElse(null);
+            } else {
+                modifiedUrl = null;
+            }
+        }
```

Comment:
This should be extracted somewhere else, it doesn't seem to be at the same level of abstraction as the rest of the code in this method. Too detailed and specific about pagination.